pkg/operator/staticpod/startupmonitor/monitor.go

```diff
@@ -106,18 +106,22 @@ func (m *monitor) Run(ctx context.Context, installerLock Locker) (ready bool, re
 	// we use wait.Until because it uses sliding interval, wait.Poll does not.
 	withTimeoutCtx, cancel := context.WithTimeout(ctx, m.timeout-m.probeInterval) // last iteration is done manually to have control over the Unlock
 	wait.UntilWithContext(withTimeoutCtx, func(ctx context.Context) {
-		defer func() {
-			if lastReady {
-				installerLock.Unlock()
-			}
-		}()
+		defer installerLock.Unlock()
 		iteration(ctx)
 		if lastReady {
 			cancel()
 		}
 	}, m.probeInterval)
+
+	// do last iteration without calling Unlock()
+	// note that you cannot logically have lastReady = true and lastError != nil
 	if !lastReady {
-		iteration(ctx) // do last iteration without Unlock
+		iteration(ctx)
+		if lastError != nil {
+			// release the lock since we are exiting anyway
+			installerLock.Unlock()
+			return false, lastReason, lastMessage, lastError
+		}
```

Comment:
this case was missing :+1:

Can we test it?

Reply:
added unit tests